OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -74,11 +74,6 @@ public override Activity Tick(Actor self)
 				self.QueueActivity(mobile.MoveTo(moveTo, 1));
 				self.SetTargetLine(Target.FromCell(self.World, moveTo), Color.Gray, false);
 
-				// TODO: The harvest-deliver-return sequence is a horrible mess of duplicated code and edge-cases
```

Comment:
The preferred value for `moveTo` is a random free cell in the vicinity of the last harvested resource cell.  It only falls back to the harvester location if there wasn't a harvested cell.